api/Service/EventService.cs

```diff
@@ -202,20 +204,25 @@ public async Task<IList<Event>> GetFutureJoinedEvents(int hostId)
 			return futureJoinedEvents;
 		}
 
-		private async Task<bool> VerifyCategory(Event myEvent)
+		private async Task<bool> VerifyCategoryHasEvents(Event myEvent)
 		{
-			//to-do:
-			//make it work with SingleOrDefaultAsync();
-			var eventsWithSameCategory = await _context.Events.Where(ev => ev.CategoryId == myEvent.CategoryId).ToListAsync();
-			return eventsWithSameCategory.Count == 1;
+			try
+			{
+				var eventsWithSameCategory = await _context.Events.Where(e => e.CategoryId == myEvent.CategoryId).SingleOrDefaultAsync();
```

Comment:
SingleOrDefault should be used instead of Where clause:
``` var eventsWithSameCategory = await _context.Events.SingleOrDefaultAsync(e => e.CategoryId == myEvent.CategoryId); ```

SingleOrDefault returns either the object (in your case an Event) or null, if the object is not found or there are multiple instances of the same object.